### PR TITLE
Tlportal 444 - log pagination headers

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

File: server/WAPI.rb [hunk 1]
```diff
@@ -149,6 +152,49 @@ def do_request(request)
...
+        prefix = "last_page: #{last_page} page_size: #{per_page} "
+      end
+      logger.warn "#{self.class.to_s}:#{__method__}:#{__LINE__}: pagination: #{prefix} next_link: #{next_link}"
+    end
+
+  end
+
+  # Extract URL for this link entry if it is the desired link type.
+  def header_link_for_rel(link, desired)
+    link_relationship = link[1][0][1]
```
Comment (zqian) [inline server/WAPI.rb]: I am confused with the data structure: link[1][0][1], and in the end, why returning link[0]?

More documentation would be appreciated!

File: server/WAPI.rb [hunk 2]
```diff
@@ -149,6 +152,49 @@ def do_request(request)
...
+      prefix = ""
+      # Log last_page and per_page values so can get rough estimate of total number of entries for query.
+      if !last_link.nil?
+        # Note: We use the page/per_page information because it appears in the URL, but that isn't a standard
+        # and it might be removed at some point.
+        p = Regexp.new(/page=(\d+)&per_page=(\d+)/)
+        p.match(last_link)
+        last_page, per_page = $1, $2
+        prefix = "last_page: #{last_page} page_size: #{per_page} "
+      end
```
Comment (zqian) [inline server/WAPI.rb]: log error if last_link is null?

File: server/WAPI.rb [hunk 3]
```diff
@@ -149,6 +152,49 @@ def do_request(request)
...
+      logger.warn "#{self.class.to_s}:#{__method__}:#{__LINE__}: pagination: #{prefix} next_link: #{next_link}"
+    end
```
Comment (zqian) [inline server/WAPI.rb]: log error if next_link is null?

File: server/WAPI.rb (hunk 1 above)
Comment (dlhaines) [inline server/WAPI.rb]: I've added some comments.

File: server/WAPI.rb (hunk 2 above)
Comment (dlhaines) [inline server/WAPI.rb]: Missing links aren't a problem.  We need to deal with them if they show up, but they can be missing.

**Pull request closed without merging**